application/controllers/UserRoleController.php

```diff
@@ -233,6 +233,10 @@ public function actionViewRole(int $ptid)
      */
     public function actionRunExport($ptid)
     {
+        if (!Permission::model()->hasGlobalPermission('superadmin', 'read')) {
+            Yii::app()->session['flashmessage'] = gT('You have no access to the role management!');
```

Comment:
403 ?

Reply:
Sorry, forgot to comment on it.
A 403 would have looked better to me as well, but the other stocks already had a flash message.

Reply:
I set as code review done